### ISSUE TITLE
Link headers

### ISSUE THREAD

**Issue by prophile** [Fyrd/caniuse#2289]
It would be interesting to see browser support for the [Link HTTP header][link].

[link]: https://tools.ietf.org/html/rfc5988

**Comment by mnot** [comment]
I think you'd need to specify what's using the Link header; e.g., stylesheets.

**Comment by nottrobin** [comment]
I see no ambiguity here - it would be lovely to know browser support for prefetching resources specified in `Link` response headers from the server:

```
Link: </images/big.jpeg>; rel=prefetch
```

As always, Mozilla explain it well:  https://developer.mozilla.org/en-US/docs/Web/HTTP/Link_prefetching_FAQ

As far as I can tell, Firefox has supported it for ages. Blink recently added support, I believe. Chrome doesn't - https://bugs.chromium.org/p/chromium/issues/detail?id=120104

**Comment by nottrobin** [comment]
See also https://www.w3.org/wiki/LinkHeader

**Comment by mnot** [comment]
It sounds like you want to include notes about whether the `Link` header is supported in addition to markup in https://caniuse.com/#feat=link-rel-prefetch

**Comment by nottrobin** [comment]
I can't quite tell right now if it's part of the same specification or not. So if you think that's the logical way to categorise it within the logic of W3C specs and of caniuse.com, then that's fair enough. As long as the data is displayed somehow, that would be good.

From a developer's perspective, however, I think they would be best served by finding information on Link HTTP header support in an independent space.

I would imagine they're rarely grouped together in a developer's head - one is HTML, the other is HTTP. One is done in template files, the other is likely in server configs. One is about building and displaying an HTML document (associated CSS etc.), the other will often be used for prefetching resources without processing them, for later use. Link HTTP header will be used more for caching and performance optimisation than building and displaying a page.

**Comment by mnot** [comment]
The relevant part of the prefetch specification is [here](https://www.w3.org/TR/resource-hints/#fetching-the-resource-hint-link); it says:

> The resource hint link's may be specified in the document markup, MAY be provided via the HTTP Link header, and MAY be dynamically added to and removed from the document.

Link headers (now specified by [RFC8288](https://httpwg.org/specs/rfc8288.html)) are part of a very broad framework for [many kinds of links](https://www.iana.org/assignments/link-relations/link-relations.xhtml), not just prefetching.

**Comment by nottrobin** [comment]
Yes I guess you're right. I had thought you could just have a single page saying whether a browser supported declaring links as HTTP headers, or not. But I guess it's probably quite likely that a browser might support `Link: <>; rel=prefetch` but not `Link: <>; rel=stylesheet` (you may know better than me). Which does rather complexify the support matrix.

So I guess you do need to either add HTTP header support as caveats to the individual support pages for each `<link>` capability, or add a new page for each `Link:` capability. Which one makes the most sense would probably depend on how often support for a capability via the `<link>` element is different from support via the `Link:` HTTP header. I'm sure you'd know which is more appropriate better than I would.
